propagation.go

```diff
@@ -45,32 +45,47 @@ var (
 type BuiltinFormat byte
 
 const (
-	// Binary encodes the SpanContext for propagation as opaque binary data.
+	// Binary represents SpanContexts as opaque binary data.
 	//
 	// For Tracer.Inject(): the carrier must be an `io.Writer`.
 	//
 	// For Tracer.Extract(): the carrier must be an `io.Reader`.
 	Binary BuiltinFormat = iota
 
-	// TextMap encodes the SpanContext as key:value pairs.
+	// TextMap represents SpanContexts as key:value string pairs.
+	//
+	// Unlike HTTPHeaders, the TextMap format does not restrict the key or
+	// value character sets in any way.
 	//
 	// For Tracer.Inject(): the carrier must be a `TextMapWriter`.
 	//
 	// For Tracer.Extract(): the carrier must be a `TextMapReader`.
+	TextMap
+
+	// HTTPHeaders represents SpanContexts as HTTP header string pairs.
 	//
-	// See HTTPHeaderTextMapCarrier for an implementation of both TextMapWriter
+	// Unlike TextMap, the HTTPHeaders format requires that the keys and values
+	// be valid as HTTP headers as-is (i.e., character casing may be unstable
+	// and special characters are disallowed in keys, values should be
+	// URL-escaped, etc).
+	//
+	// For Tracer.Inject(): the carrier must be a `TextMapWriter`.
+	//
+	// For Tracer.Extract(): the carrier must be a `TextMapReader`.
+	//
+	// See HTTPHeaderCarrier for an implementation of both TextMapWriter
 	// and TextMapReader that defers to an http.Header instance for storage.
 	// For example, Inject():
 	//
-	//    carrier := HTTPHeaderTextMapCarrier(httpReq.Header)
-	//    err := span.Tracer().Inject(span, TextMap, carrier)
+	//    carrier := HTTPHeadersCarrier(httpReq.Header)
+	//    err := span.Tracer().Inject(span, HTTPHeaders, carrier)
```

Comment:
nit: probably should be using package name in the examples, `opentracing.HTTPHeaders`